lib/streamlit/watcher/local_sources_watcher.py

```diff
@@ -61,22 +61,30 @@ def __init__(self, pages_manager: "PagesManager"):
         self.update_watched_pages()
 
     def update_watched_pages(self) -> None:
-        old_watched_pages = self._watched_pages
+        old_page_paths = self._watched_pages.copy()
         new_pages_paths: set[str] = set()
```

Comment:
Hm, I might be missing something about the desired behavior here, but what's the reason we only deregister watchers when a page file is no longer valid? I doubt it matters too much since the behavioral change will only be noticeable in dev mode, but it seems like we'll often end up having watchers on pages not being displayed in the case that the pages shown by `st.navigation` are dynamically changing.

Reply:
Yea, it's an edge case, but feels...technically the right setup. Handles two cases

1. Changes to files with multiple people using the app. pages can update.
2. People can dynamically create pages now and pass it to `st.navigation` and delete them as well.